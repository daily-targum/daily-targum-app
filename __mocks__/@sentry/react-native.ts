export const init = () => jest.fn();