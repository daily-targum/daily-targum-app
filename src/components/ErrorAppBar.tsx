import React, { useEffect, useRef } from 'react';
import { Text } from 'react-native';
import { useDispatch } from 'react-redux';
import { errorActions, useErrorSelector } from '../store/ducks/error';
import Theme from '../components/Theme';
import Animated, { Easing } from 'react-native-reanimated';

const HEIGHT = 30;

function ErrorSnackbar() {
  const {colors, insets} = Theme.useTheme();
  const error = useErrorSelector(s => s.error);
  const dispatch = useDispatch();

  let id: NodeJS.Timeout;
  useEffect(() => {
    id = setTimeout(() => {
      dispatch(errorActions.clearError());
    }, 5000);
    return () => clearTimeout(id);
  }, [error]);

  const visible = error !== null;
  const height = useRef(new Animated.Value(0)).current;
  useEffect(() => {
    Animated.timing(height, {
      toValue: +visible,
      duration: 250,
      easing: Easing.inOut(Easing.ease),
    }).start();
  }, [visible]);

  return (
    <Animated.View
      style={{
        height: height.interpolate({
          inputRange: [0, 1],
          outputRange: [0, HEIGHT + insets.top],
        }),
        backgroundColor: colors.accent,
        alignItems: 'center',
        justifyContent: 'center'
      }}
    >
      <Text
        style={{
          color: '#fff',
          paddingTop: insets.top
        }}
      >{error}</Text>
    </Animated.View>
  );
}

export default ErrorSnackbar;