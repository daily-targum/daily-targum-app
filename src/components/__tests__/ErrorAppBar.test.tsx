jest.mock('../Theme');
jest.useFakeTimers();

import React from 'react';
import ErrorAppBar from '../ErrorAppBar';
import renderer from 'react-test-renderer';
import { Provider as ReduxProvider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Text } from 'react-native';

const mockStore = configureStore([]);

describe('<ErrorAppBar />', () => {

  it('without error', () => {
    const store = mockStore({
      error: {
        error: null
      }
    });
    const tree = renderer.create(
      <ReduxProvider store={store}>
        <ErrorAppBar/>
      </ReduxProvider>
    );
    renderer.act(() => {});
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.root.findByType(Text).props.children).toBe(null);
  });

  it('with error', () => {
    const ERROR = '404. Page Not Found.';
    const store = mockStore({
      error: {
        error: ERROR
      }
    });
    const tree = renderer.create(
      <ReduxProvider store={store}>
        <ErrorAppBar/>
      </ReduxProvider>
    );
    renderer.act(() => {});
    expect(tree.root.findByType(Text).props.children).toBe(ERROR);
  });

});