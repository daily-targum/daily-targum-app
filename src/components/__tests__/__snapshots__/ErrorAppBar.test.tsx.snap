// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorAppBar /> without error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f70737",
      "height": undefined,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "paddingTop": 30,
      }
    }
  />
</View>
`;
